src/heatmap/HeatMapImageRenderer.ts

```diff
@@ -130,11 +134,47 @@ export default class HeatMapImageRenderer extends AHeatMapCanvasRenderer impleme
     } else if (colors.length === 2 || colors.length === 3) {
       args.palette = colors.map(ensureHex).join('-');
     }
-    this.image.src = data.heatmapUrl(all(), args);
+
+    // persist to get range and create range object again
+    // TODO: make range property on matrix public
+    const range = parseRange(data.persist().range);
+    const params = prepareParameter(range, args);
+    const url = `/dataset/matrix/${data.desc.id}/data`;
+
+    const encoded = encodeParams(params);
+    if (encoded && (url.length + encoded.length > TOO_LONG_URL)) {
+      // use post instead
+      sendAPI(url, params, 'POST', 'blob').then((image) => {
+        const imageURL = window.URL.createObjectURL(image);
+        this.image.src = imageURL;
+      });
+    } else {
+      this.image.src = data.heatmapUrl(all(), args);
```

Comment:
This is also called when `encoded` results to `false` no matter how long the URL is. Is this intended?

Reply:
the function `encodeParams(data: any = null)` returns a string, therefore encoded can only be empty, `null` or a string

Reply:
But in the if statement you write `if (encoded && (url.length + encoded.length > TOO_LONG_URL)) {` when `encoded === False` then automatically choose the GET option even if the url is longer than 4096. Is this intended?